packages/plugin-core/src/test/suite-integ/OpenLink.test.ts

```diff
@@ -1,31 +1,100 @@
 import { vault2Path } from "@dendronhq/common-server";
 import { ENGINE_HOOKS } from "@dendronhq/engine-test-utils";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
 import fs from "fs-extra";
 import path from "path";
 import * as vscode from "vscode";
 import { OpenLinkCommand } from "../../commands/OpenLink";
-import { expect } from "../testUtilsv2";
+import { expect, runSingleVaultTest } from "../testUtilsv2";
 import { runLegacyMultiWorkspaceTest, setupBeforeAfter } from "../testUtilsV3";
+import { VSCodeUtils } from "../../utils";
+import { NoteProps } from "@dendronhq/common-all";
 
 suite("OpenLink", function () {
-  let ctx: vscode.ExtensionContext;
-  ctx = setupBeforeAfter(this);
+  const ctx = setupBeforeAfter(this, {});
 
-  test("error: nothing selected", (done) => {
+  test("error: cursor on non-link", (done) => {
     runLegacyMultiWorkspaceTest({
       ctx,
       preSetupHook: async ({ wsRoot, vaults }) => {
         ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
       },
-      onInit: async ({}) => {
+      onInit: async () => {
         const cmd = new OpenLinkCommand();
         const { error } = await cmd.execute();
-        expect(error!.message).toEqual("nothing selected");
+        expect(error!.message).toEqual("no valid path or URL selected");
         done();
       },
     });
   });
 
+  //TODO
+  test.skip("With an invalid character in the selection.", (done) => {
+    let noteWithLink: NoteProps;
+    runSingleVaultTest({
```

Comment:
`runSingleVaultTest` requires a different setup. you want to use `runLegacyMultiWorkspaceTest` and then tests will work